src/services/AuthService.ts

```diff
@@ -0,0 +1,25 @@
+import jwt, { JwtPayload } from "jsonwebtoken";
+import { secrets } from "../utils/secrets";
+
+const jwtPrivateKey = `-----BEGIN EC PRIVATE KEY-----\n${secrets.JWTPrivateKey}\n-----END EC PRIVATE KEY-----`;
+const jwtPublicKey = `-----BEGIN PUBLIC KEY-----\n${secrets.JWTPublicKey}\n-----END PUBLIC KEY-----`;
+const algorithm = "ES256";
+
+export class AuthService {
+  public decodeToken(token: string): JwtPayload {
+    if (token.toLowerCase().startsWith("bearer"))
+      token = token.slice("bearer".length);
+
+    return jwt.verify(token.trim(), jwtPublicKey, {
+      complete: true,
+      algorithms: [algorithm],
+    });
+  }
+
+  public generateToken(username: string): string {
```

Comment:
Especially this one could confuse a lot of people as `AuthService.generateToken` is very similar to `TokenService.generateTokens`.

Maybe rename the former to `generateJWT` 🤷‍♂️.